src/main/java/org/junit/runner/manipulation/SortBasedOrdering.java

```diff
@@ -0,0 +1,38 @@
+package org.junit.runner.manipulation;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.runner.Description;
+
+class SortBasedOrdering extends Ordering {
+    private final Sorter sorter;
+
+    public SortBasedOrdering(Sorter sorter) {
+        this.sorter = sorter;
+    }
+
+    @Override
+    public void apply(Object runner)  {
+        if (runner instanceof Sortable) {
```

Comment:
This was some of the ugly code that caused me to want to make `Sorter` an `Ordering`.